### ISSUE TITLE
Start partition number calculation

### ISSUE THREAD

**Issue by red-led** [keithf4/pg_partman#98]
Hi!

Trying to run ```create_parent``` on existing table with lots of data I [found](https://github.com/keithf4/pg_partman/blob/master/sql/functions/create_parent.sql#L390-L395) it trys to run this: 
```sql
SELECT COALESCE('0', max('id')::bigint, 0) FROM "Schema"."Table" LIMIT 1
```
where ```'0'``` is value ```p_start_partition``` parameter. In my case this query executes in about 1.5-2 minutes. In combination with [access exclusive lock](https://github.com/keithf4/pg_partman/blob/master/sql/functions/create_parent.sql#L94) made before it creates some troubles. Is it really need to run this query when non-null ```p_start_partition``` is passed? If yes, maybe it's posstble to run this before locking table?

**Comment by keithf4** [comment]
Your right, I could probably move the attempt for the lock after that. I'll dig into it further to make sure. And could also likely avoid that query if that start parameter is set.

Thanks for the suggestion!

**Comment by keithf4** [comment]
This has been fixed in version 2.3.2. Thanks for the report and assistance with fixing it!
